src/pybind/mgr/cephadm/ssh.py

```diff
@@ -0,0 +1,136 @@
+import asyncssh
+import asyncssh.logging
+import logging
+import os
+from io import StringIO
+from shlex import quote
+from typing import TYPE_CHECKING, Optional, List, Tuple
+from orchestrator import OrchestratorError
+
+if TYPE_CHECKING:
+    from cephadm.module import CephadmOrchestrator
+    from asyncssh import SSHClientConnection
+
+logger = logging.getLogger(__name__)
+
+
+class SSHConnection:
+
+    cons = {}
+
+    def __init__(self, mgr: "CephadmOrchestrator"):
+        self.mgr: "CephadmOrchestrator" = mgr
+
+    async def _remote_connection(self,
```

Comment:
I think I'm starting to feel your pain here. I wonder if we could build a function that looks like so:

```python
@contextmanager
def remote_connection
    try:
        conn = CephadmServe.loop.run_until_complete(self.ssh._remote_connection(host, addr))
        yield conn
    except ...
```
Would that have any benefit?

Reply:
hmm I can try it out, but I'm not sure if that would make it simpler. i can revisit if we need